src/main/java/com/google/sps/data/DeliverySlotManager.java

```diff
@@ -44,14 +45,14 @@
     private double latitude;
     private double longitude;
 
-    public CreateSlotManager(String deliveryDay, String timezoneOffsetMinutes, String startTime, String endTime, String maxStops) throws BadRequestException {
+    public CreateSlotManager(String deliveryDay, String timezoneOffsetMinutes, String start, String end, String maxStops) throws BadRequestException {
       this.timezoneOffset = parseInteger(timezoneOffsetMinutes) * 60;
-      int startTimeSeconds = getNumberOfSeconds(parseTime(startTime)) + this.timezoneOffset;
-      int endTimeSeconds = getNumberOfSeconds(parseTime(endTime)) + this.timezoneOffset;
-      if (startTimeSeconds > endTimeSeconds) {
+      Time startTime = Time.valueOf(parseTime(start));
```

Comment:
In general you want to handle input conversions in the servlet, before you call into your other classes.

That way your classes don't each have to deal with user raw strings. Even if it doesn't make a difference now, in the long run, you'll see a lot of the parsing code duplicated a across a lot of classes, which introduces more room for error.

Also your internal classes (like SlotManager) shouldn't really be dealing with user input (not a great separation of responsibilities).

Reply:
This makes a lot of sense, the only reason why I added the parsing code in CreateSlotManager.java was to test that the input is parsed correctly. 
If I move it back into the servlet, is it okay to only test that the request is valid (e.g. the startTime is not set after endTime)